department-of-veterans-affairs/caseflow
# Add veterans waiting in virtual hearings queue

Resolves https://vajira.max.gov/browse/CASEFLOW-389

### Description

- Add `R` for virtual hearing RO in `REGIONAL_OFFICE_INFORMATION.json` file (this resembles how we use `C` for the central office RO)
- Update various places to handle a hearing day with a virtual request type
- Relax validation to support a virtual hearing day without an RO
- Add handling to the assign hearings tab to find tasks with virtual hearing requests
- Update regional office dropdown to show `Virtual Hearings` label instead of city and state 
- Update tests and add new ones

### Acceptance Criteria
- [x] Work is enabled only with the `national_vh_queue` feature flag
- [x] The new "Virtual Hearings" option is available in the Regional Office dropdown of the Schedule Veterans page
- [x] When a user selects the "Virtual Hearings" option
  - [x] the Legacy Veterans Waiting tab shows a list of all legacy appeals with a hearing request type of "Virtual"
  - [x] the AMA Veterans Waiting tab shows a list of all AMA appeals with a hearing request type of "Virtual"
- [x] Any appeals that appear in the Legacy Veterans Waiting tab or AMA Veterans Waiting tab also appear in their geo-matched RO queue
- [x] Find any other places that regional office JSON is being used and filtered to make sure Virtual Hearings addition doesn't show up where it shouldn't.

### Testing Plan
1.  Setup a new virtual hearing day by running this code in console:
    ```
    pry(main)> HearingDay.last.dup.update!(request_type: 'R', regional_office: nil)
    ```
2. Navigate to the Schedule Veterans page...you should see the "Virtual Hearings" option toward the bottom. If you don't, you might need to enable the `national_vh_queue` toggle
3. Try converting a travel board hearing to virtual: http://localhost:3000/queue/appeals/1986897
4. Ensure that the hearing shows as ready to schedule in both "Virtual Hearings" and (probably) St. Petersburg
5. Ensure that the "Virtual Hearings" option doesn't display when you try to schedule a veteran

### User Facing Changes

<img width="1424" alt="Screen Shot 2021-01-12 at 6 13 57 PM" src="https://user-images.githubusercontent.com/1298274/104385836-f180a400-5501-11eb-9782-4c7b54fa594e.png">
